src/mds/Server.cc

```diff
@@ -1615,7 +1654,26 @@ void Server::submit_mdlog_entry(LogEvent *le, MDSLogContextBase *fin, MDRequestR
 void Server::respond_to_request(MDRequestRef& mdr, int r)
 {
   if (mdr->client_request) {
-    reply_client_request(mdr, MClientReply::create(*mdr->client_request, r));
+    MClientReply::ref reply = MClientReply::create(*mdr->client_request, r);
+    reply_client_request(mdr, reply);
+    if (mdr->is_batch_op() && mdr->is_batch_head) {
+      int mask = mdr->client_request->head.args.getattr.mask;
+
+      BatchOpContext *fin;
+      if (mdr->client_request->get_op() == CEPH_MDS_OP_GETATTR) {
+	fin = mdr->in[0]->batch_ops[mask];
```

Comment:
auto it = mdr->in[0]->batch_ops.find(mask);
if (it != mdr->in[0]->batch_ops.end()) {
fin = it->second;
mdr->in[0]->batch_ops.erase(mask);
}

Reply:
@ukernel Hi, thanks for your review:-)
When the execution of the code reaches here, batch_ops is guaranteed to have the callback for the "mask", since the callback is added to batch_ops right before is_batch_head is set to true:-)

Reply:
using find() is more efficient.